common/djangoapps/student/helpers.py

```diff
@@ -270,6 +262,31 @@ def get_next_url_for_login_page(request):
     return redirect_to
 
 
+def get_redirect_to(request):
+    """
+    Determine the redirect url and return if safe
+    :argument
+        request: request object
+
+    :returns: redirect url if safe else None
+    """
+    static_file_format = '\w+\.(jpeg|jpg|gif|png|ico)'
```

Comment:
My question wasn't answered. I think we're digging ourselves into a pretty deep hole here without thinking about the bigger picture. Why is it a problem if a user is redirected to an image or other static asset hosted on our site?

As you mentioned, the goal is to prevent a malicious third party from redirecting users to their site. `http.is_safe_url()` fixes that problem. Why do we need to worry about adding a whitelist on top of that to deal with static assets on our site?